api/v1alpha1/project_config_types.go

```diff
@@ -13,13 +24,42 @@ type ProjectConfig struct {
 	metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
 	// Spec describes the configuration of a Project.
 	Spec ProjectConfigSpec `json:"spec,omitempty" protobuf:"bytes,2,opt,name=spec"`
+	// Status describes the current status of a Project Config.
+	Status ProjectConfigStatus `json:"status,omitempty" protobuf:"bytes,3,opt,name=status"`
+}
+
+func (p *ProjectConfig) GetStatus() *ProjectConfigStatus {
+	return &p.Status
 }
 
 // ProjectConfigSpec describes the configuration of a Project.
 type ProjectConfigSpec struct {
 	// PromotionPolicies defines policies governing the promotion of Freight to
 	// specific Stages within the Project.
 	PromotionPolicies []PromotionPolicy `json:"promotionPolicies,omitempty" protobuf:"bytes,1,rep,name=promotionPolicies"`
+	// WebhookReceiverConfigs defines the webhook receivers for the project config.
+	WebhookReceiverConfigs []WebhookReceiverConfig `json:"webhookReceivers,omitempty" protobuf:"bytes,2,rep,name=receivers"`
+}
+
+type ProjectConfigStatus struct {
+	// Conditions contains the last observations of the Project Config's current state.
+	// +patchMergeKey=type
+	// +patchStrategy=merge
+	// +listType=map
+	// +listMapKey=type
+	Conditions []metav1.Condition `json:"conditions,omitempty" patchMergeKey:"type" patchStrategy:"merge" protobuf:"bytes,3,rep,name=conditions"`
+	// WebhookReceivers contains the list of webhook receivers for the project config.
+	WebhookReceivers []WebhookReceiver `json:"webhookReceivers,omitempty" protobuf:"bytes,5,rep,name=receivers"`
```

Comment:
Refer to offline convo we had. If you omit protobuf struct tags when adding a new field, codegen will add them for you when you run it. In this case, codegen would have used 1 and 2 instead of 3 and 5.